src/theory/arrays/theory_arrays.cpp

```diff
@@ -395,7 +395,8 @@ bool TheoryArrays::propagate(TNode literal)
 }/* TheoryArrays::propagate(TNode) */
 
 
-void TheoryArrays::explain(TNode literal, std::vector<TNode>& assumptions, eq::EqProof *proof) {
+void TheoryArrays::explain(TNode literal, std::vector<TNode>& assumptions,
```

Comment:
It seems like `std::shared_ptr<EqProof> eqp` should be a `std::weak_ptr` since `explain` is not taking (shared) ownership.